compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -298,6 +298,7 @@ public SemanticAnalyzer(CompilerContext context) {
     }
 
     public BLangPackage analyze(BLangPackage pkgNode) {
+        this.prevEnvs.clear();
```

Comment:
Ideally we shouldn't have this line, but still `envs` for TestablePackages and Packages are getting accumulated when there are multiple compilations. Hope this is fine. @MaryamZi @hasithaa